pkg/util/helper/binding.go

```diff
@@ -45,19 +48,18 @@ func (p ClusterWeightInfoList) Less(i, j int) bool {
 	if p[i].Weight != p[j].Weight {
 		return p[i].Weight > p[j].Weight
 	}
-	return p[i].ClusterName < p[j].ClusterName
-}
-
-// SortClusterByWeight sort clusters by the weight
-func SortClusterByWeight(m map[string]int64) ClusterWeightInfoList {
-	p := make(ClusterWeightInfoList, len(m))
-	i := 0
-	for k, v := range m {
-		p[i] = ClusterWeightInfo{k, v}
-		i++
-	}
-	sort.Sort(p)
-	return p
+	// when weights is equal, sort by last scheduling replicas result,
+	// more last scheduling replicas means the remainders of the last scheduling were randomized to such clusters,
+	// so in order to keep the inertia in this scheduling, such clusters should also be prioritized
+	if p[i].LastReplicas != p[j].LastReplicas {
+		return p[i].LastReplicas > p[j].LastReplicas
+	}
+	// when last scheduling replicas is also equal, sort by random,
+	// first generate a random number within [0, 100) range,
+	// then return < if the actual number is in [0, 50) range, return > if is in [50, 100) range
+	const maxRandomNum = 100
+	randomNum, err := rand.Int(rand.Reader, big.NewInt(maxRandomNum))
+	return err == nil && randomNum.Cmp(big.NewInt(maxRandomNum/2)) >= 0
```

Comment:
What's the difference between the random algorithm with `math/rand`? 
```golang
	mathrand.New(mathrand.NewSource(time.Now().UnixNano()))
	randi := mathrand.Int()
	randj := mathrand.Int()
	return randi > randj
```

Reply:
There's not much difference, just to achieve a randomness

Reply:
shall I change to this method ?

Reply:
shall I change to your approch ? @RainbowMango

Reply:
Yeah, I get it, thanks for the reminder. `match/rand` package might bring unnecessary concerns, especially security scanners.